src/app/sb-admin/components/frameworkmgmt/term/term.component.ts

```diff
@@ -0,0 +1,151 @@
+import { Component, OnInit } from '@angular/core';
+import { Subscription } from 'rxjs';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { Message, MessageService } from 'primeng/api';
+import { I18NextPipe } from 'angular-i18next';
+
+import { UserService } from 'src/app/sb-admin/service/user.service';
+import { SearchFilterValue } from 'src/app/sb-admin/interfaces/user';
+import { CategoryName, CategoryCode } from 'src/config/constant.config';
+
+@Component({
+  selector: 'app-term',
+  templateUrl: './term.component.html',
+  styleUrls: ['./term.component.scss']
+})
+export class TermComponent implements OnInit {
+  private subscription!: Subscription;
+  createTerm!: FormGroup;
+  submitted = false;
+  messages: Message[] = [];
+  organizations: any[] = [];
+  frameworks: any[] = [];
+  selectedOrg: any[] = [];
+  filteredValue = SearchFilterValue;
+  filtered1Value = SearchFilterValue;
+  first = 0;
+  orgId: any;
+  CategoryName = CategoryName;
+  CategoryCode = CategoryCode;
+  node: any;
+
+  constructor(
+    private userService: UserService,
+    private messageService: MessageService,
+    public formBuilder: FormBuilder,
+    private i18nextPipe: I18NextPipe
+  ) {}
+
+  ngOnInit() {
+    this.initializeAddForm();
+    this.getOrganizations();
+  }
+
+  initializeAddForm() {
+    this.createTerm = this.formBuilder.group({
+      filteredValue: [null, Validators.required],
+      frameworkName: [null, Validators.required],
+      categoryName: ['', Validators.required],
+      termCode: ['', Validators.required],
+      termName: ['', Validators.required],
+      termLabel: ['', Validators.required],
+      description: ['', Validators.required]
+    });
+  }
+
+  saveTerm() {
+    this.submitted = true;
+    const updatedFormValues = { ...this.createTerm.value };
+    this.mapCategoryNames(updatedFormValues);
+
+    const body = {
+      request: {
+        term: {
+          name: updatedFormValues.termName,
+          label: updatedFormValues.termLabel,
+          description: updatedFormValues.description,
+          code: updatedFormValues.termCode
+        }
+      }
+    };
+   this.subscription= this.userService.createTerm(body, updatedFormValues).subscribe(
+      (response) => {
+        this.node = response.result.node_id;
+        this.messages = [];
+        this.messageService.add({
+          severity: 'success',
+          detail: this.i18nextPipe.transform('FRAMEWORK_ADDED')
+        });
+        this.createTerm.reset();
+        this.filteredValue = null;
+        this.submitted = false;
+      },
+      (error) => {
+        this.handleCategoryCreationError(error);
+      }
+    );
+  }
+// Maps human-readable category names to their corresponding system identifiers.
+  mapCategoryNames(updatedFormValues: any): void {
+    if (updatedFormValues.categoryName === "Grade") {
```

Comment:
Define constants for BGMS

Reply:
Changed the mapping function and declared the BMGS as const